pattern_import_export/models/ir_exports.py

```diff
@@ -16,133 +15,134 @@ class IrExports(models.Model):
                 }
             )
+            attachment.unlink()
         return True
 
-
-class IrExportsLine(models.Model):
-    _inherit = "ir.exports.line"
-
-    select_tab_id = fields.Many2one("ir.exports.select.tab", string="Select tab")
-    is_many2x = fields.Boolean(
-        string="Is Many2x field", compute="_compute_is_many2x", store=True
-    )
-    related_model_id = fields.Many2one(
-        "ir.model",
-        string="Related model",
-        compute="_compute_related_model_id",
-        store=True,
-    )
-
-    def _get_last_field(self, model, path):
-        if "/" not in path:
-            path = path + "/"
-        field, path = path.split("/", 1)
-        if path:
-            model = self.env[model]._fields[field]._related_comodel_name
-            return self._get_last_field(model, path)
-        else:
-            return field, model
-
     @api.multi
-    @api.depends("name")
-    def _compute_is_many2x(self):
-        for export_line in self:
-            if export_line.export_id.resource and export_line.name:
-                field, model = export_line._get_last_field(
-                    export_line.export_id.resource, export_line.name
-                )
-                if self.env[model]._fields[field].type in ["many2one", "many2many"]:
-                    export_line.is_many2x = True
+    def _get_data_to_export(self, records):
+        """
+        Iterator who built data dict record by record
+        @param records: recordset
+        @return: dict
+        """
+        self.ensure_one()
+        headers = self._get_header()
+        for record in records:
+            data = {}
+            fields_to_export = []
+            column_headers = {}
+            for export_line in self.export_fields:
+                nb_column = 1
+                if export_line.is_many2x and export_line.select_tab_id:
+                    field_name = export_line.select_tab_id.field_id.name
+                    field = export_line.name + "/" + field_name
+                    if export_line.is_many2many:
+                        nb_column = max(1, export_line.number_occurence)
+                else:
+                    field = export_line.name
+                for _i in range(0, nb_column):
+                    fields_to_export.append(field)
+                column_headers.update({field: nb_column})
+            # The data-structure returned by export_data is different
+            # that the one used to export.
+            # export_data(...) return a dict with a keys named 'datas'
+            # and it contains a list of list.
+            # Each list item is a line (for M2M) but for the export,
+            # we want to display these lines as column.
+            # So we have to convert
+            res = record.export_data(fields_to_export, raw_data=False)
+            for target_ind, record_data in enumerate(res.get("datas", []), start=1):
+                for header, value in zip(headers, record_data):
```

Comment:
I will do something more like.

Line: 80
```
if export_line.is_many2many:
   m2m_fields[export_line.name] = export_line.occurence
```
Then here something like

```
data = zip(fields_to_export, record_data)
for key in m2m_fields:
    vals = data.pop(key)
    for idx in range(0, nb_column):
        try:
            data[key + " | " + idx] = vals[idx]
        except IndexError:
             data[key + " | " + idx] = None
```
        
What do you think?

Reply:
Sorry but it couldn't work. The `zip(...)` doesn't have a `pop` attribute and the data-structure returned by `record.export_data(...)` is not so simple. When you have more than 1 `O2M` or `M2M` fields to export, your code won't work.
I just test some modifications based on your code but it becomes more complicated (as it's now).